crates/cli/src/subcommands/publish.rs

```diff
@@ -367,11 +349,35 @@ async fn apply_pre_publish_if_needed(
                     println!("{}", manual.reason);
                     println!("Proceeding with database clear due to --delete-data=always.");
                 }
+                println!(
+                    "This will DESTROY the current {} module, and ALL corresponding data.",
+                    name_or_identity
+                );
+                if !y_or_n(
+                    force,
+                    format!("Are you sure you want to proceed? [deleting {}]", name_or_identity).as_str(),
+                )? {
+                    anyhow::bail!("Aborting");
+                }
+                builder = builder.query(&[("clear", true)]);
             }
             PrePublishResult::AutoMigrate(auto) => {
+                if clear_database == ClearMode::Always {
+                    println!("Auto-migration, does NOT require clearing the database, but proceeding with database clear due to --delete-data=always.");
```

Comment:
this case doesn't seem to be tested? If I remove this block and run `cargo test publish`, they all still seem to pass.

My default position is that we should do all of the tests described in https://github.com/clockworklabs/SpacetimeDB/pull/3730#pullrequestreview-3494666487:

- no migration required
  - delete-data=no
  - delete-data=on-conflict
  - delete-data=yes
- auto-migrateable
  - delete-data=no, yes-break=false
  - delete-data=no, yes-break=true
  - delete-data=on-conflict, yes-break=false
  - delete-data=on-conflict, yes-break=true
  - delete-data=yes, yes-break=false
  - delete-data=yes, yes-break=true
- manual migration required
  - delete-data=no
  - delete-data=on-conflict
  - delete-data=yes

Reply:
unfortunately they still would not catch this case, because the tests do not check whether or not data was actually deleted.